### ISSUE TITLE
Add zap stanza for Paragon Casks

### ISSUE THREAD

**Issue by RogerThiede** [caskroom/homebrew-cask#16458]
References #7327

The current uninstall stanzas for cask paragon-ntfs and cask paragon-extfs leave some files and folders on the system in preparation for a re-install later.  A zap stanza would be needed to eliminate these files and folders.

```
"~/Library/Application Support/Paragon NTFS for Mac OS X/.notfirsttime"
"~/Library/Preferences/com.paragon-software.filesystems.ntfs.plist"
"~/Library/Preferences/com.paragon-software.NTFS.prefpane.plist"
"/Library/Application Support/Paragon Software/NTFS10"
"/Library/Application Support/Paragon Software/NTFS12"
"/Library/Application Support/Paragon Software/NTFS14"
```
```
"~/Library/Application Support/Paragon ExtFS for Mac OS X/.notfirsttime"
"~/Library/Preferences/com.paragon-software.filesystems.extfs.plist"
"/Library/Application Support/Paragon Software/ExtFS9"
```
It is not clear to me why paragon-ntfs references two similar preference pane plists; I expect one of them is from an older version of paragon-ntfs.

Should a zap stanza remove all known instances of historical installations, or should a zap stanza only touch files that the most recent known ```brew cask install paragon-ntfs``` would create?

**Comment by vitorgalvao** [comment]
> or should a zap stanza only touch files that the most recent known `brew cask install paragon-ntfs` would create?

That one. Otherwise we’d have a confusing ever-expanding list.

**Comment by vitorgalvao** [comment]
Care to submit the PR?

**Comment by RogerThiede** [comment]
I plan to submit a PR, however I just realized that if a zap stanza only touched files from the most recent known ``brew cask install paragon-ntfs``, then files installed by older versions have an opportunity to never be removed.

For example, the use case:
- Fresh System
- brew cask install paragon-ntfs
- Paragon creates a new version such as changing from NTFS14 to NTFS15
- Developers update paragon-ntfs.rb to install newer version as well as zapping only newer version
- brew update
- User still has NTFS14 installed and performs brew cask zap paragon-ntfs
- User is left with stale NTFS14 file.
- brew cask install paragon-ntfs

**Comment by vitorgalvao** [comment]
You are correct. However, that is a limitation of the system as it works currently. In the future, we’d like to have glob support (i.e. `/Library/Application Support/Paragon Software/NTFS*` would consider (from your examples) `10`, `12`, `14`.

As we don’t support it yet, we have to keep only the newer versions. You can [do something like this](https://github.com/caskroom/homebrew-cask/blob/c76c1b219c44f1feca4a3776f62a54f46a4dbd61/Casks/gpgtools.rb#L57), though. It won’t work for any version currently, but will be ready for when the feature lands.

**Comment by haodong** [comment]
```
brew cask install paragon-ntfs
==> Downloading http://dl.paragon-software.com/demo/ntfsmac14_trial_e.dmg
Already downloaded: /.../Homebrew/paragon-ntfs-14.dmg
==> Verifying checksum for Cask paragon-ntfs
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 9b2a2e5b0b2eec51ef85aa72048a9da7aa3eab3a07477a3e670dca31cd7a4ad9
Actual: 2bdb44c0cf5feaf3e5619a737fef849d0f171b782784f3e2315e50bf6c68ce12
File: /.../Homebrew/paragon-ntfs-14.dmg
To retry an incomplete download, remove the file above.
```

**Comment by adityadalal924** [comment]
@haodong Next time, [open up an new issue](https://github.com/caskroom/homebrew-cask#reporting-bugs). Your error is unrelated.

Fixed in #16474

**Comment by vitorgalvao** [comment]
No need to keep this open anymore. Question has been answered.
